### PR TITLE
Nix shell sync with Substrate

### PR DESCRIPTION
**Pull request by kaichaosun**
Also fixed error when using macos.
https://github.com/paritytech/substrate/pull/7376

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+let
+  mozillaOverlay =
+    import (builtins.fetchGit {
+      url = "https://github.com/mozilla/nixpkgs-mozilla.git";
+      rev = "57c8084c7ef41366993909c20491e359bbb90f54";
+    });
+  nixpkgs = import <nixpkgs> { overlays = [ mozillaOverlay ]; };
+  rust-nightly = with nixpkgs; ((rustChannelOf { date = "2020-10-01"; channel = "nightly"; }).rust.override {
```
Comment (riusricardo) [inline shell.nix:8]: Is it possible to use the same version as docker, 2020-10-05?

**Review by riusricardo — COMMENTED** [review]
Will it work if we move it to the scripts folder?

**Comment by riusricardo** [comment]
Ok, I just saw your other PR. It needs to be in the root folder.

File: shell.nix (hunk 1 above)
Comment (pepyakin) [reply shell.nix:8]: meanwhile it was already upgraded : )

File: shell.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+in
+with nixpkgs; pkgs.mkShell {
+  buildInputs = [
+    clang
+    cmake
+    pkg-config
+    rust-nightly
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
```
Comment (pepyakin) [inline shell.nix:20]: care to upstream this?

File: shell.nix (hunk 1 above)
Comment (kaichaosun) [reply shell.nix:8]: node-template still depends on v2.0.0, i believe latest working nightly is 2020-10-05.

**Pull request merged**